File: src/main/java/modules/Reactions.java
package modules;

import base.Command;
import base.EschaUtil;
import base.Module;
import discord4j.common.util.Snowflake;
import discord4j.core.GatewayDiscordClient;
import discord4j.core.event.domain.message.MessageCreateEvent;
import discord4j.core.object.entity.Guild;
import discord4j.core.object.entity.Message;
import discord4j.core.object.entity.User;
import discord4j.core.object.entity.channel.Channel;
import discord4j.core.object.entity.channel.TextChannel;
import reactor.core.publisher.Mono;

import java.util.*;

public class Reactions extends Module {
    private String[] enk = {
            "wtf|why are you awake",
            "why are you awake",
            "shouldn't you be asleep",
            "shouldn't you be studying",
            ":gotosleep:",
            "wtf|:gotosleep:",
            "its past your bedtime",
            "isnt it past your bedtime|why are you awake"
    };

    public Reactions(GatewayDiscordClient client) {
        super(client, "");

        this.client.on(MessageCreateEvent.class)
                .flatMap(event -> Mono.justOrEmpty(event.getMessage())
                        .filter(message -> message.getAuthor().map(user -> !user.isBot()).orElse(false))
                        .flatMap(message -> message.getChannel()
                                .ofType(TextChannel.class)
                                .flatMap(channel -> checkForReaction(event)))).subscribe();
    }

    private Mono<Void> checkForReaction(MessageCreateEvent event) {
        Guild guild = event.getGuild().block();
        Channel channel = event.getMessage().getChannel().block();
        if (ChannelPerms.canModuleIn(guild, getName(), channel)) {
            Message message = event.getMessage();
            User author = message.getAuthor().get();
            String msg = message.getContent();
            Set<Snowflake> mentioned = message.getUserMentionIds();

            if (msg.replace("\'", "").toLowerCase().contains("dont quote me")) {
                return EschaUtil.sendMessage(event, "\"" + msg + "\" -" + author.getUsername());
            } else if (Arrays.stream(msg.toLowerCase().split(" ")).anyMatch(s -> s.equalsIgnoreCase("alot"))) {
                return EschaUtil.sendMessage(event, "http://thewritepractice.com/wp-content/uploads/2012/05/Alot-vs-a-lot1-600x450.png");
            } else if (Arrays.stream(msg.toLowerCase().split(" ")).anyMatch(s -> s.equalsIgnoreCase("lambo"))) {
                return EschaUtil.sendMessage(event, "https://streamable.com/qxx1ob");
            } else if (author.getId().asLong() == 207986006840836097L && mentioned.contains(Snowflake.of(102559179507519488L))) {
//            } else if (author.getId().asLong() == 85844964633747456L && mentioned.contains(Snowflake.of(216037525754609664L))) {
                int choice = new Random().nextInt(enk.length);
                this.client.getGuilds()
                        .filter(g -> g.getId().asLong() == 85847666780553216L)
                        .flatMap(emojiGuild -> emojiGuild.getGuildEmojiById(Snowflake.of(667622392612257842L))
//                this.client.getGuilds()
//                        .filter(g -> g.getId().asLong() == 259163710109646849L)
//                        .flatMap(emojiGuild -> emojiGuild.getGuildEmojiById(Snowflake.of(667623052309168128L))
                                .flatMap(gotosleep -> {
                                    String[] lines = enk[choice].split("\\|");
                                    for (String s : lines) {
                                        if (s.contains(":gotosleep:"))
                                            s = s.replace(":gotosleep:", gotosleep.asFormat());
                                        EschaUtil.sendMessage(event, s).subscribe();
                                    }
                                    return Mono.empty();
                                })).subscribe();
                return Mono.empty();
            }
        }
        return Mono.empty();
    }

    @Override
    protected Map<String, Command> makeCommands() {
        return Collections.emptyMap();
    }

    @Override
    public String getName() {
        return "Reactions";
    }
}

File: cogs/reactions.py
import re
import random
from discord import DMChannel
from discord.ext import commands


class Reactions(commands.Cog):
    """Random situations the bot reacts to"""

    def __init__(self, bot):
        self.ENK = [
            "wtf|why are you awake",
            "why are you awake",
            "shouldn't you be asleep",
            "shouldn't you be studying",
            ":gotosleep:",
            "wtf|:gotosleep:",
            "its past your bedtime",
            "isnt it past your bedtime|why are you awake"
        ]

        self.bot = bot

    @commands.Cog.listener()
    async def on_message(self, msg):
        if isinstance(msg.channel, DMChannel):
            return
        if not self.bot.utils.can_cog_in(self, msg.channel) and msg.author.id != self.bot.user.id:
            return
        if msg.author.bot:
            return
        await self.check_for_reaction(await self.bot.get_context(msg), msg)

    async def check_for_reaction(self, ctx, msg):
        m = msg.content.strip()
        split = [word.lower() for word in re.split(' |\n', m)]
        if 'dont quote me' in m.replace('\'', '').lower():
            await ctx.send(f'"{m.strip()}" - {ctx.author.mention}')
        elif 'alot' in split:
            await ctx.send('https://cdn.discordapp.com/attachments/356568034279686146/1174092586210828410/Alot-vs-a-lot1-600x450.png?ex=656655ee&is=6553e0ee&hm=2f2fed1a99ecb3b82a112e8c0891c72c2b44b19b8c6c9d45e075c5a8aa09f24e&')
        elif 'lambo' in split:
            await ctx.send('https://youtu.be/6w1TieX_I2w')
        # elif msg.author.id == 85844964633747456 and 131547909090050048 in [u.id for u in msg.mentions]:
        elif msg.author.id == 207986006840836097 and 102559179507519488 in [u.id for u in msg.mentions]:
            choice = random.randint(0, len(self.ENK) - 1)
            lines = self.ENK[choice].split('|')
            for l in lines:
                await ctx.send(l.replace(':gotosleep:', '<a:gotosleep:667623052309168128>'))
        elif m.startswith('/o/'):
            await ctx.send('\o\\')
        elif m.startswith('\o\\'):
            await ctx.send('/o/')


async def setup(bot):
    await bot.add_cog(Reactions(bot))
